control-plane-operator/controllers/hostedcontrolplane/ingress/ingresscontroller.go

```diff
@@ -0,0 +1,82 @@
+package ingress
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/config"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/util"
+)
+
+var (
+	operatorV1Scheme = runtime.NewScheme()
+)
+
+func init() {
+	operatorv1.AddToScheme(operatorV1Scheme)
+}
```

Comment:
I don't think you need this stuff, as we have a [general purpose global scheme](https://github.com/openshift/hypershift/blob/main/api/scheme.go#L39) in the api package already to use

Reply:
I thought about that and realized that these are api groups that we never deal with directly on the management cluster. We're only using these to serialize stuff that will be applied to the guest cluster. Should we really be adding these to the same schema that's used for the hypershift operator? It may make sense to have a separate schema for the control plane operator.

Reply:
Hrm, good point, and now that you've made it I wonder if each operator should have its own scheme to work with. I don't want to inadvertently turn the api package into a shared dumping ground. How about starting a new scheme in a single shared place in the the control plane operator to use then rather than putting it in this specific component file?

Reply:
Yup, setting up a new shared schema for the control plane operator. Will also clean up the one in the api package to remove things we don't directly use on the management side.